### PR TITLE
TF destroy fix

### PR DESCRIPTION
**Pull request by raghavanandan**
Description:
---------------
This PR is to address follow-ups and potential fixes in the sysdig provider resource for GCP API only onboarding.

### CODE REVIEW

**Review by mdkulkarni15 — APPROVED** [review]
Looks good to me.

File: sysdig/resource_sysdig_secure_cloud_auth_account.go [hunk 1]
```diff
@@ -396,19 +396,92 @@ func cloudauthAccountFromResourceData(data *schema.ResourceData) *v2.CloudauthAc
...
+/*
+	This helper function converts the features data from *cloudauth.AccountFeatures to resource data schema.
+	This is needed to set the value in cloudauthAccountToResourceData().
+*/
+
+func featureToResourceData(features *cloudauth.AccountFeatures) []map[string]interface{} {
+	featureMap := []map[string]interface{}{}
+
+	featureFields := map[string]*cloudauth.AccountFeature{
+		"secure_threat_detection":     features.SecureThreatDetection,
```
Comment (mdkulkarni15) [inline sysdig/resource_sysdig_secure_cloud_auth_account.go:428]: all these strings in quotes, here and below, should we have them defined as constants above?

**Review by ravinadhruve10 — APPROVED** [review]

File: sysdig/resource_sysdig_secure_cloud_auth_account.go [hunk 2]
```diff
@@ -16,20 +16,43 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
 )
 
+const (
```
Comment (mdkulkarni15) [inline sysdig/resource_sysdig_secure_cloud_auth_account.go:19]: you can put them in sysdig/common.go like other constants.

File: sysdig/resource_sysdig_secure_cloud_auth_account.go [hunk 3]
```diff
@@ -382,33 +405,106 @@ func constructAccountComponents(accountComponents []*cloudauth.AccountComponent,
...
+		if feature != nil {
+			value := featureValuesToResourceData(name, feature)
+			featureMap = append(featureMap, value)
 		}
 	}
+
+	return featureMap
+}
+
+func cloudauthAccountToResourceData(data *schema.ResourceData, cloudAccount *v2.CloudauthAccountSecure) error {
```
Comment (cgeers) [inline sysdig/resource_sysdig_secure_cloud_auth_account.go:468]: I wish we hadn't blown the implementation of this function from the terse version we previously had.

**Pull request merged**